src/conn/conn_prefetch.c

```diff
@@ -98,21 +98,24 @@ __wt_prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
             break;
         }
         __wt_spin_lock(session, &conn->prefetch_lock);
-        locked = true;
         pe = TAILQ_FIRST(&conn->pfqh);
 
         /* If there is no work for the thread to do - return back to the thread pool */
-        if (pe == NULL)
+        if (pe == NULL) {
+            __wt_spin_unlock(session, &conn->prefetch_lock);
             break;
+        }
 
         TAILQ_REMOVE(&conn->pfqh, pe, q);
         --conn->prefetch_queue_count;
 
+        ((WT_BTREE *)pe->dhandle->handle)->prefetch_busy += 1;
```

Comment:
If this is only ever 1 or 0 we should convert it to a boolean.

Reply:
Without an atomic increment, this value can go backward. A similar problem was observed with `btree->write_gen` and later converted into an atomic increment.

Reply:
Okay so I'll convert this back to an atomic load and store.